docs: add `WITHIN GROUP` for `string_agg` & `array_agg` (#1683)
databendlabs/databend-docs: docs/en/sql-reference/20-sql-functions/07-aggregate-functions/aggregate-string-agg.md

Aggregate function.

> The STRING_AGG() function (also known by its alias GROUP_CONCAT or LISTAGG) converts all the non-NULL values of a column to String, separated by the delimiter.

Reviewer: ```suggestion
The STRING_AGG() function (also known as GROUP_CONCAT or LISTAGG) converts all the non-NULL values of a column to a string, separated by the delimiter.

```

|-------------|---------------------------------------------------------------------|
| `<expr>`    | Any string expression (if not a string, use `::VARCHAR` to convert) |
| `delimiter` | Optional constant String, if not specified, use empty String        |
| `<expr1>`   | Any expression                                                      |

## Optional

| Optional                            | Description                                            |
|-------------------------------------|--------------------------------------------------------|

> | WITHIN GROUP &lt;orderby_clause&gt; | defines the order of values for ordered set aggregates |

Reviewer: As above.
